sympy/functions/combinatorial/numbers.py

```diff
@@ -834,9 +834,16 @@ class catalan(Function):
 
     @classmethod
     def eval(cls, n):
-        if n.is_Integer and n.is_nonnegative:
+        if (n.is_Integer and n.is_nonnegative) or \
+           (n.is_noninteger and n.is_negative):
             return 4**n*C.gamma(n + S.Half)/(C.gamma(S.Half)*C.gamma(n + 2))
 
+        if (n.is_integer and n.is_negative):
+            if (n + 1).is_negative:
+                return S.Zero
+            if n == -S.One:
```

Comment:
I think if you use `==` you can just write `-1`; or use ` is S.NegativeOne` if you are testing for a literal Integer. Or perhaps you mean `(n + 1).is_zero`?